### PR TITLE
feat: support jump to workspace file with position

### PR DESCRIPTION
**Pull request by njzydark**
### Description

https://github.com/user-attachments/assets/d6c3f523-ecfa-49b5-bc31-cf8bae474683

### Additional context

The logic for calculating the position might be somewhat hacky

### CODE REVIEW

File: src/data.ts [hunk 1]
```diff
@@ -64,6 +90,98 @@ export class PnpmWorkspaceManager {
...
+    let namedStartPosition: [number, number] = [0, 0]
+    let custimzedNamePosition: [number, number] = [0, 0]
+
+    let data: PnpmWorkspacPositionData = {
+      catalog: undefined,
+      catalogs: undefined
+    }
+
+    let currentCatalogName = ''
+    for (let lineIndex = 0; lineIndex < doc.lineCount; lineIndex++) {
```
Comment (antfu) [inline src/data.ts:112]: Should we use a proper YAML parser instead of regex?

Comment (njzydark) [reply src/data.ts:112]: I haven't found a suitable library; the one I'm currently using, js-yaml, doesn't support it

Comment (antfu) [reply src/data.ts:112]: Maybe try https://github.com/ota-meshi/yaml-eslint-parser?

Comment (njzydark) [reply src/data.ts:112]: Thank you, I will look into it later.

**Pull request closed without merging**